### PR TITLE
catalog/resource_grants: Refactor to use Go SDK

### PR DESCRIPTION
**Pull request by martin-walsh**
## Changes

Follow on from #3024 and update resource_grants to use Go SDK.

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by martin-walsh** [comment]
See comment re `materialized_view` & `view` securable as I cannot see these documented on https://docs.databricks.com/api/workspace/grants/update

File: catalog/resource_grants.go [hunk 1]
```diff
@@ -308,7 +241,7 @@ var mapping = securableMapping{
 		"APPLY_TAG":      true,
 		"EXECUTE":        true,
 	},
-	"materialized_view": {
+	"materialized_view": { // TODO: Is this supported as it is not documented ?
```
Comment (nkvuong) [inline catalog/resource_grants.go:244]: you are correct - this is done via `table` securable

File: catalog/resource_grants.go [hunk 2]
```diff
@@ -210,7 +143,7 @@ var mapping = securableMapping{
 		"APPLY_TAG":      true,
 		"BROWSE":         true,
 	},
-	"view": {
+	"view": { // TODO: Is this supported as it is not documented ?
```
Comment (nkvuong) [inline catalog/resource_grants.go:146]: you are correct - this is done via `table` securable

File: internal/acceptance/grant_test.go [hunk 3]
```diff
@@ -54,6 +54,13 @@ resource "databricks_external_location" "some" {
 	comment         = "Managed by TF"
 }
 
+resource "databricks_grant" "metastore" {
```
Comment (nkvuong) [inline internal/acceptance/grant_test.go:57]: could you comment this block out for now (we need to exclude this from running in parallel with the grants test first)

Comment (martin-walsh) [reply internal/acceptance/grant_test.go:57]: Sure.

File: catalog/resource_grants.go (hunk 2 above)
Comment (martin-walsh) [reply catalog/resource_grants.go:146]: Ok, so, should I keep the `view` for backwards compat and just pass it to `table` similarly to `model` --> `function`, or remove these args ?.

Ditto `materialized_view` above.

Comment (martin-walsh) [reply catalog/resource_grants.go:146]: Follow up question: The newly added `databricks_grant` does not support these as I followed the api docs, so if you choose to keep `view` as an arg here should I add it to `databricks_grant` also for consistency ?

File: internal/acceptance/grant_test.go (hunk 3 above)
Comment (martin-walsh) [reply internal/acceptance/grant_test.go]: Done

File: catalog/resource_grants.go (hunk 2 above)
Comment (nkvuong) [reply catalog/resource_grants.go:146]: `view` and `materialized_view` have never worked (the API will error out), so should be safe to remove - you just need to add these permissions back to `table` mapping

Comment (martin-walsh) [reply catalog/resource_grants.go]: Removed

**Review by mgyucht — APPROVED** [review]
Thanks for this!

@nkvuong let's talk offline about that, I can address it as I'm on-call this week.

**Pull request merged**